File: Math/876/inverse_element.cpp
#include <iostream>
using namespace std;

int n;
long long a, p, res;

long long fast_power(long long a, long long b, long long p);

int main() {
    scanf("%d", &n);
    while (n--) {
        scanf("%d%d", &a, &p);
        if (!(a % p)) printf("impossible\n");
        else printf("%d\n", fast_power(a, p - 2, p));
    }
    return 0;
}

long long fast_power(long long a, long long b, long long p) {
    long long res = 1;
    while (b) {
        if (b & 1) res = (res * a) % p;
        a = (a * a) % p;
        b >>= 1;
    }
    return res;
}

File: Math/876/inverse_element.py
def binpow(a: int, b: int, p: int) -> int:
    res = 1
    while b:
        if b & 1:
            res = res * a % p
        a = a * a % p
        b >>= 1
    return res
    
if __name__ == "__main__":
    n = int(input())
    for _ in range(n):
        a, p = map(int, input().split())
        if a % p == 0:
            print("impossible")
        else:
            print(binpow(a, p - 2, p))
